### PR TITLE
`General`: Add "Dismiss all" button to alerts overlay

### PR DESCRIPTION
**Pull request by bassner**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] Following the [theming guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

It is currently not possible to dismiss all alerts at once. Instead, a user has to click the little "X" in the top right of each alert, which can be time-consuming, especially if you quickly want to access content covered by alerts. 
This was a suggestion in my thesis that I wanted to cover after I finished, and it should solve #5630 well enough.

### Description

Added a button at the top of the alert stack that allows to dismiss all alerts at once. It is only visible if there is more than one alert; if there is only one alert, one can use the X button with the same effort and the new button would just take up space unnecessarily.

### Steps for Testing

Prerequisites:
- 1 User

1. Provoke the app to spawn multiple alerts at once. I recommend testing this locally, and just adding something like this somewhere into the code: 
```js
setTimeout(() => this.alertService.info('hi'), 1000);
setTimeout(() => this.alertService.success('Test'), 2000);
```
2. Verify that you can dismiss all alerts at once if you click the button.
3. Verify that the button only appears if the are multiple active alerts
4. Verify that the button disappears if you close all alerts but one.

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

https://user-images.githubusercontent.com/23171488/197762849-aaa34fa8-d801-4526-a9f9-8040ebdf1b7e.mov

### CODE REVIEW

File: src/main/webapp/app/shared/alert/alert-overlay.component.html [hunk 1]
```diff
@@ -10,3 +9,7 @@
         <div class="time-bar" *ngIf="(alert?.timeout || 0) > 0" [ngStyle]="{ animationDuration: alert.timeout + 'ms' }"></div>
     </div>
 </div>
+<div class="btn btn-secondary dismiss-all" *ngIf="(alerts?.length ?? 0) >= 2" (click)="alertService.closeAll()" @dismissAllAnimation>
```
Comment (simon-hng) [inline src/main/webapp/app/shared/alert/alert-overlay.component.html:12]: 🧠 cool use of nullish coalescing

**Review by simon-hng — DISMISSED** [review]
code good

**Review by simon-hng — DISMISSED** [review]
Code even better now

**Review by JohannesStoehr — DISMISSED** [review]
I'd put the dismiss all button at the top of all alerts. In the unlikely case of so many alerts that the screen is filled, you can still access it, and it would always be at the same place regardless of the amount of notifications (might be annoying if an additional pops up or an old one disappears)

**Comment by simon-hng** [comment]
I think having the dismiss button on top is rather uncommon. A better way would be to alerts scrollable IMO

**Comment by bassner** [comment]
Moved the button to the top of the stack, @JohannesStoehr :)

**Review by ge65cer — APPROVED** [review]
Tested on TS1, nice addition!

**Review by simon-hng — APPROVED** [review]

**Pull request merged**